main.go

```diff
@@ -298,7 +300,7 @@ func check_handler(w http.ResponseWriter, r *http.Request) {
 					logged = true
 				}
 			}
-			if logged == false {
+			if !logged {
```

Comment:
One thing I think you might interest you is the [logrus](https://github.com/sirupsen/logrus) library. You may already know, but it'll let you log at different levels of granularity. This will free you from having to make explicit logging checks and leave you to have to only worry about the actual business logic you really care about.

Reply:
This logged variable is checking for past payments. Its not for debugging but you're probably right in that I should have proper logging.